CMakeLists.txt

```
@@ -50,18 +57,28 @@ file(GLOB SOURCES "edlib/src/*.cpp")
 
 # Create libraries.
 add_library(edlib ${SOURCES})
+add_library(edlib::edlib ALIAS edlib)
 set_target_properties(edlib
-    PROPERTIES
+  PROPERTIES
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN ON
     VERSION ${PROJECT_VERSION}
-    SOVERSION ${PROJECT_VERSION})
+    SOVERSION ${PROJECT_VERSION}
+)
+
+if(BUILD_SHARED_LIBS)
+  target_compile_definitions(edlib PUBLIC EDLIB_SHARED PRIVATE EDLIB_BUILD_SHARED)
+endif()
```

See documentation for target property [`DEFINE_SYMBOL`](https://cmake.org/cmake/help/latest/prop_tgt/DEFINE_SYMBOL.html).

```suggestion
    DEFINE_SYMBOL EDLIB_BUILD_SHARED
    SOVERSION ${PROJECT_VERSION}
)

if(BUILD_SHARED_LIBS)
  target_compile_definitions(edlib PUBLIC EDLIB_SHARED)
endif()
```